quantum/matrix.c

```diff
@@ -67,28 +67,33 @@ static bool read_cols_on_row(matrix_row_t current_matrix[], uint8_t current_row)
 }
 
 #elif defined(DIODE_DIRECTION)
-#    if (DIODE_DIRECTION == COL2ROW)
-
-static void select_row(uint8_t row) {
-    setPinOutput(row_pins[row]);
-    writePinLow(row_pins[row]);
-}
-
-static void unselect_row(uint8_t row) { setPinInputHigh(row_pins[row]); }
 
 static void unselect_rows(void) {
     for (uint8_t x = 0; x < MATRIX_ROWS; x++) {
         setPinInputHigh(row_pins[x]);
     }
 }
 
-static void init_pins(void) {
-    unselect_rows();
+static void unselect_cols(void) {
     for (uint8_t x = 0; x < MATRIX_COLS; x++) {
         setPinInputHigh(col_pins[x]);
     }
 }
 
+static void init_pins(void) {
+    unselect_cols();
+    unselect_rows();
+}
```

Comment:
The only difference that I can see is previously there was a difference of the order of calling, for `ROW2COL` it was

```c
unselect_cols();
unselect_rows();
```

And for `COL2ROW` it was

```c
unselect_rows();
unselect_cols();
```